src/inmanta/deploy/persistence.py

```diff
@@ -163,8 +163,12 @@ async def send_in_progress(self, action_id: UUID, resource_id: Id) -> None:
                 except UniqueViolationError:
                     raise ValueError(f"A resource action with id {action_id} already exists.")
 
-                # FIXME: we may want to have this in the RPS table instead of Resource table, at some point
+                # FIXME: At some point, we will only need this on the RPS table
                 await resource.update_fields(connection=connection, status=const.ResourceState.deploying)
+                await resource.update_persistent_state(
+                    is_deploying=False,
```

Comment:
```suggestion
                    is_deploying=True,
```

Reply:
This also suggests we are not testing this. I would try to add some assertions to existing tests (that may already test consistency)

Reply:
:facepalm: Maybe this is why on my makeshift tests I've never got it to work